src/pybamm/util.py

```diff
@@ -123,10 +123,18 @@ def search(self, keys, print_values=False):
             If True, print both keys and values. Otherwise, print only keys.
             Default is False.
         """
+
+        if not keys or (isinstance(keys, str) and not keys.strip()):
+            print("Error: Please provide a non-empty search term.")
+            return
+        if isinstance(keys, list) and all(not str(k).strip() for k in keys):
+            print("Error: Please provide a non-empty search term in the list.")
+            return
+
         if isinstance(keys, str):
-            search_keys = [keys]
+            search_keys = [keys.strip()]
         else:
-            search_keys = keys.copy()
+            search_keys = [k.strip() for k in keys if k.strip()]
```

Comment:
We should perform these operations after creating a copy, i.e., leave `original_keys` intact, and then move these to line 140 and after.